platform/platform-impl/src/com/intellij/openapi/project/DumbServiceImpl.java

```diff
@@ -326,6 +328,10 @@ private void enterDumbMode(@Nullable TransactionId contextTransaction, @NotNull
 
   private void queueUpdateFinished() {
     if (myState.compareAndSet(State.RUNNING_DUMB_TASKS, State.WAITING_FOR_FINISH)) {
+      // There is no tasks to suspend with the current suspender. If the execution reverts to the dumb mode, a new suspender will be
+      // created. The suspender might have already got suspended between the point of the last check cancelled call and this point, however.
+      // It if has happened it will be cleaned up when the suspender is closed on the background process thread.
```

Comment:
small typo

Reply:
I updated the comment to make more clear, but I'm not sure what you refer to.

Reply:
"it if" should probably be "if it"

Reply:
sure, updated